File: DigiMox/app/src/main/java/com/digimox/adapters/DMFeedbackAdapter.java
package com.digimox.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.LinearLayout;
import android.widget.RatingBar;
import android.widget.TextView;

import com.digimox.R;
import com.digimox.models.response.DMFeedback;

import java.util.ArrayList;

/**
 * Created by Deepesh on 21-Dec-15.
 */
public class DMFeedbackAdapter extends BaseAdapter {

    private static final int CURSOR_TEXT_COLUMN = 0;
    private ArrayList<DMFeedback> dmFeedbacks;
    private Context context;
    private ViewHolder holder = null;
    public ArrayList<String> rates;

    public DMFeedbackAdapter(Context context, ArrayList<DMFeedback> dmFeedbacks) {
        this.dmFeedbacks = dmFeedbacks;
        this.context = context;
        rates = new ArrayList<>(dmFeedbacks.size());
        for (int i = 0; i < dmFeedbacks.size(); i++) {
            rates.add(i, String.valueOf(0));
        }

    }

    @Override
    public Object getItem(int position) {
        return dmFeedbacks.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup viewGroup) {
        LayoutInflater mInflater = LayoutInflater.from(context);

        if (convertView == null) {
            convertView = mInflater.inflate(R.layout.layout_feedback_item, viewGroup,
                    false);
            holder = new ViewHolder();
            holder.text = (TextView) convertView.findViewById(R.id.feedback_question);
            holder.ratingCount = (TextView) convertView.findViewById(R.id.star_count);
            holder.progress = (RatingBar) convertView
                    .findViewById(R.id.feedback_rating);
            holder.progress.setTag(position);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }
        holder.text.setText(dmFeedbacks.get(position).getFeedbackLangQuestion());
        holder.progress
                .setOnRatingBarChangeListener(new RatingBar.OnRatingBarChangeListener() {

                    @Override
                    public void onRatingChanged(RatingBar ratingBar,
                                                float rating, boolean fromUser) {
                        Integer myPosition = (Integer) ratingBar.getTag();
                        rates.set(myPosition, String.valueOf((int) rating));
                        LinearLayout parent = (LinearLayout) ratingBar.getParent();
                        TextView label = (TextView) parent.findViewById(R.id.star_count);
                        label.setText(String.valueOf((int) rating));
                    }
                });
        return convertView;
    }


    @Override
    public int getCount() {
        return dmFeedbacks.size();
    }

    static class ViewHolder {
        TextView text;
        TextView ratingCount;
        RatingBar progress;
    }

}

File: DigiMox/app/src/main/java/com/digimox/adapters/DMFeedbackAdapter.kt
package com.digimox.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.LinearLayout
import android.widget.RatingBar
import android.widget.TextView
import com.digimox.R
import com.digimox.models.response.DMFeedback
import java.util.*

/**
 * Created by Deepesh on 21-Dec-15.
 */
class DMFeedbackAdapter(private val context: Context, private val dmFeedbacks: ArrayList<DMFeedback>) : BaseAdapter() {
    private var holder: ViewHolder? = null
    var rates: ArrayList<String> = ArrayList(dmFeedbacks.size)

    init {
        for (i in dmFeedbacks.indices) {
            rates.add(i, 0.toString())
        }

    }

    override fun getItem(position: Int): Any {
        return dmFeedbacks[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, viewGroup: ViewGroup): View {
        var convertView = convertView
        val mInflater = LayoutInflater.from(context)

        if (convertView == null) {
            convertView = mInflater.inflate(R.layout.layout_feedback_item, viewGroup,
                    false)
            holder = ViewHolder()
            holder!!.text = convertView!!.findViewById<View>(R.id.feedback_question) as TextView
            holder!!.ratingCount = convertView.findViewById<View>(R.id.star_count) as TextView
            holder!!.progress = convertView
                    .findViewById<View>(R.id.feedback_rating) as RatingBar
            holder!!.progress!!.tag = position
            convertView.tag = holder
        } else {
            holder = convertView.tag as ViewHolder
        }
        holder!!.text!!.text = dmFeedbacks[position].feedbackLangQuestion
        holder!!.progress!!.onRatingBarChangeListener = RatingBar.OnRatingBarChangeListener { ratingBar, rating, fromUser ->
            val myPosition = ratingBar.tag as Int
            rates[myPosition] = rating.toInt().toString()
            val parent = ratingBar.parent as LinearLayout
            val label = parent.findViewById<View>(R.id.star_count) as TextView
            label.text = rating.toInt().toString()
        }
        return convertView
    }


    override fun getCount(): Int {
        return dmFeedbacks.size
    }

    internal class ViewHolder {
        var text: TextView? = null
        var ratingCount: TextView? = null
        var progress: RatingBar? = null
    }

    companion object {

        private val CURSOR_TEXT_COLUMN = 0
    }

}